collectors/scm/github/src/main/java/com/capitalone/dashboard/collector/DefaultGitHubClient.java

```diff
@@ -177,6 +184,264 @@ public DefaultGitHubClient(GitHubSettings settings,
+		if (repo.getBranch() != null)
+			branch = repo.getBranch();
+		String pageUrl = apiUrl.concat("/pulls?state=all&base="+branch);
+
+		// decrypt password
+		String decryptedPassword = "";
+		if (repo.getPassword() != null && !repo.getPassword().isEmpty()) {
+			try {
+				decryptedPassword = Encryption.decryptString(
+						repo.getPassword(), settings.getKey());
+			} catch (EncryptionException e) {
+				LOG.error(e.getMessage());
+			}
+		}
+
+		try {
+			ResponseEntity<String> response = makeRestCall(pageUrl, repo.getUserId(), decryptedPassword);
+
+			HttpHeaders headers = response.getHeaders();
+			List<String> pagevalues = headers.get("Link");
+			int pageCount = 0;
+
+			/*
+<https://github.kdc.capitalone.com/api/v3/repositories/11436/pulls?state=all&base=master&page=2>; rel="next",
+<https://github.kdc.capitalone.com/api/v3/repositories/11436/pulls?state=all&base=master&page=20>; rel="last"			 */
+			if (pagevalues == null) {
+				pageCount = 1;
+			} else {
+				String[] splited1 = pagevalues.get(0).split(";");
+				String[] splited2 = splited1[1].split("=");
+				String[] splitted3 = splited2[4].split(">");
+				String lastPageCount = splitted3[0];
+				pageCount = Integer.parseInt(lastPageCount);
+			}
+
+			int pageNumber = 1;
+			while (pageNumber <= pageCount) {
+				try {
+					String queryUrl = pageUrl + "&page=" + pageNumber;
+					LOG.info("Executing [" + queryUrl);
+					response = makeRestCall(queryUrl, repo.getUserId(), decryptedPassword);
+					JSONArray jsonArray = paresAsArray(response);
+					for (Object item : jsonArray) {
+						JSONObject jsonObject = (JSONObject) item;
+						String message = str(jsonObject, "title");
+						String number = str(jsonObject, "number");
+
+						JSONObject userObject = (JSONObject) jsonObject.get("user");
+						String name = str(userObject, "login");
+						String created = str(jsonObject, "created_at");
+						String merged = str(jsonObject, "merged_at");
+						String closed = str(jsonObject, "closed_at");
+						long createdTimestamp = new DateTime(created).getMillis();
+
+						Pull pull = new Pull();
+
+						if (merged != null && merged.length() >= 10) {
+							long mergedTimestamp = new DateTime(merged).getMillis();
+							pull.setScmCommitTimestamp(mergedTimestamp);
+							pull.setResolutiontime((mergedTimestamp - createdTimestamp) / (24 * 3600000));
+						}
+						pull.setUserId(name);
+						pull.setScmUrl(repo.getRepoUrl());
+						pull.setTimestamp(createdTimestamp);
+						pull.setScmRevisionNumber(number);
+						pull.setScmCommitLog(message);
+						pull.setCreatedAt(created);
+						pull.setClosedAt(closed);
+						pull.setMergedAt(merged);
+						pull.setNumber(number);
+						String reponameArray[] = pageUrl.split("/");
+
+						Pull preExistingPull = pullRepository.findByOrgNameAndRepoNameAndNumber(
+								reponameArray[6], reponameArray[7], number);
+						if ( preExistingPull!= null) {
+							pullRepository.delete(preExistingPull);
```

Comment:
Field number (parameter location) as I parse the git repository name provided in widget. For readability, I will change that into static final consts